applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c

```diff
@@ -398,7 +432,176 @@ static struct bt_bap_stream_ops stream_ops = {
 #endif /* (CONFIG_BT_AUDIO_TX) */
 };
 
-static int initialize(le_audio_receive_cb recv_cb, le_audio_timestamp_cb timestmp_cb)
+int unicast_server_config_get(uint32_t *bitrate, uint32_t *sampling_rate_hz,
+			      uint32_t *pres_delay_us)
+{
+	if (bitrate == NULL && sampling_rate_hz == NULL && pres_delay_us == NULL) {
+		LOG_ERR("No valid pointers received");
+		return -ENXIO;
+	}
+
+	if (audio_streams[0].codec == NULL) {
+		LOG_ERR("No codec found for the stream");
+		return -ENXIO;
+	}
+
+	/* Get the configuration for the sink stream */
+	int frames_per_sec = 1000000 / bt_codec_cfg_get_frame_duration_us(audio_streams[0].codec);
+	int bits_per_frame = bt_codec_cfg_get_octets_per_frame(audio_streams[0].codec) * 8;
+
+	if (sampling_rate_hz != NULL) {
+		*sampling_rate_hz = bt_codec_cfg_get_freq(audio_streams[0].codec);
+	}
+
+	if (bitrate != NULL) {
+		*bitrate = frames_per_sec * bits_per_frame;
+	}
+
+	if (pres_delay_us != NULL) {
+		if (audio_streams[0].qos == NULL) {
+			LOG_ERR("No QoS found for the stream");
+			return -ENXIO;
+		}
+
+		*pres_delay_us = audio_streams[0].qos->pd;
+	}
+
+	return 0;
+}
+
+int unicast_server_adv_get(struct net_buf_simple *uuid_buf, struct bt_data *adv_buf,
+			   uint8_t adv_buf_vacant)
+{
+	int adv_buf_cnt = 0;
+
+	if ((adv_buf_vacant - adv_buf_cnt) == 0) {
```

Comment:
Should create a static helper function for this

Reply:
Agree. This will likely be the case for several modes of operation. Suggest we take this after the four major PRs have been merged.